gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/sideEffect/GroupSideEffectStep.java

```diff
@@ -77,7 +77,7 @@ public void modulateBy(final Traversal.Admin<?, ?> kvTraversal) {
 
     @Override
     protected void sideEffect(final Traverser.Admin<S> traverser) {
-        final Map<K, Object> map = new HashMap<>();
+        final Map<K, Object> map = new HashMap<>(1);
```

Comment:
That line of code is as you have it in master/. Was done in another PR. Also, the load factor can't be 1.0 as that map will presumably grow on reduction.